### PR TITLE
Need help with creating a `Distribution` class

### PR DESCRIPTION
**Pull request by larryshamalama**
The file outlined in this PR attempts to create a `Distribution` class for stick-breaking weights for the Dirichlet process. Weights are generated as followed:

Screen Shot 2021-07-12 at 10 00 57 AM

for some alpha > 0 value. However, the code yields the following error:

```python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-5-07ccfca52ddd> in <module>
      1 with pm.Model() as model:
----> 2     sbw = StickBreakingWeights("test-sticks", alpha=1)
      3 
      4     trace = pm.sample(1000)
      5     print(trace.to_dict()["posterior"]["test-sticks"][0].mean())

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/pymc3/distributions/distribution.py in __new__(cls, name, rng, dims, initval, observed, total_size, transform, *args, **kwargs)
    205         # Create the RV without specifying initval, because the initval may have a shape
    206         # that only matches after replicating with a size implied by dims (see below).
--> 207         rv_out = cls.dist(*args, rng=rng, initval=None, **kwargs)
    208         ndim_actual = rv_out.ndim
    209         resize_shape = None

<ipython-input-2-b7a30db55ce6> in dist(cls, alpha, *args, **kwargs)
     35         assert_negative_support(alpha, "alpha", "StickBreakingWeights")
     36 
---> 37         return super().dist([alpha], **kwargs)
     38 
     39     def logp(value, alpha):

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/pymc3/distributions/distribution.py in dist(cls, dist_params, shape, size, initval, **kwargs)
    283         # Create the RV with a `size` right away.
    284         # This is not necessarily the final result.
--> 285         rv_out = cls.rv_op(*dist_params, size=create_size, **kwargs)
    286         rv_out = maybe_resize(
    287             rv_out,

<ipython-input-2-b7a30db55ce6> in __call__(self, alpha, size, **kwargs)
      7 
      8     def __call__(self, alpha, size=None, **kwargs):
----> 9         return super().__call__(alpha, size=size, **kwargs)
     10 
     11     @classmethod

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in __call__(self, size, name, rng, dtype, *args, **kwargs)
    321 
    322     def __call__(self, *args, size=None, name=None, rng=None, dtype=None, **kwargs):
--> 323         res = super().__call__(rng, size, dtype, *args, **kwargs)
    324 
    325         if name is not None:

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/graph/op.py in __call__(self, *inputs, **kwargs)
    269         """
    270         return_list = kwargs.pop("return_list", False)
--> 271         node = self.make_node(*inputs, **kwargs)
    272 
    273         if config.compute_test_value != "off":

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in make_node(self, rng, size, dtype, *dist_params)
    368             raise TypeError("The type of rng should be an instance of RandomStateType")
    369 
--> 370         bcast = self.compute_bcast(dist_params, size)
    371         dtype = self.dtype or dtype
    372 

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/configparser.py in res(*args, **kwargs)
     47         def res(*args, **kwargs):
     48             with self:
---> 49                 return f(*args, **kwargs)
     50 
     51         return res

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in compute_bcast(self, dist_params, size)
    285 
    286         """
--> 287         shape = self._infer_shape(size, dist_params)
    288 
    289         # Ignore `Cast`s, since they do not affect broadcastables

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in _infer_shape(self, size, dist_params, param_shapes)
    253             ndim_reps = len(shape_reps)
    254         else:
--> 255             shape_supp = self._shape_from_params(
    256                 dist_params,
    257                 param_shapes=param_shapes,

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in _shape_from_params(self, dist_params, **kwargs)
    155         Defaults to `param_supp_shape_fn`.
    156         """
--> 157         return default_shape_from_params(self.ndim_supp, dist_params, **kwargs)
    158 
    159     def rng_fn(self, rng, *args, **kwargs):

~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/random/op.py in default_shape_from_params(ndim_supp, dist_params, rep_param_idx, param_shapes)
     68     if param_shapes is not None:
     69         ref_param = param_shapes[rep_param_idx]
---> 70         return (ref_param[-ndim_supp],)
     71     else:
     72         ref_param = dist_params[rep_param_idx]

IndexError: tuple index out of range
```

Any thoughts are appreciated!

### CODE REVIEW

File: test-dp-submodule/stick-breaking-weights-rv.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+from aesara.tensor.random.op import RandomVariable
+
+from pymc3.distributions.continuous import assert_negative_support
+from pymc3.distributions.distribution import Continuous
+
+
+class StickBreakingWeightsRV(RandomVariable):
+    name = "stick_breaking_weights"
+    ndim_supp = 1
```
Comment (Sayam753) [inline test-dp-submodule/stick-breaking-weights-rv.py:16]:
```suggestion
    ndim_supp = 0
```
Can you try this and see if this works?

I think with `ndim_support = 1`, RandomVariable class is expecting the `alpha` parameter to be a vector of 1 dimension

Comment (larryshamalama) [reply test-dp-submodule/stick-breaking-weights-rv.py:16]: I think that this is correct... I tried most combinations of `ndim_supp = 0, ndims_params = [1]`, etc., but ultimately I believe that `alpha` needs to be a vector. I am getting a new error, but at least I am able to get to the NUTS initialization step 😅

Comment (larryshamalama) [reply test-dp-submodule/stick-breaking-weights-rv.py:16]: Thanks so much!!

**Comment by larryshamalama** [comment]
[Link](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/random/basic.py#L346) to `DirichletRV` from `aesara` for reference

**Pull request merged**